reconnecting-eventsource.d.ts

```diff
@@ -0,0 +1,21 @@
+export interface Configuration extends EventSourceInit {
+    // the maximum time to wait before attempting to reconnect in ms, default `3000`
+    // note: wait time is randomised to prevent all clients from attempting to reconnect simultaneously
+    max_retry_time: number;
+}
+
+export default class ReconnectingEventSource {
+    readonly url: string;
+    readonly readyState: number;
+    max_retry_time: number;
+
+    constructor (url: string, configuration?: Configuration);
+
+    onopen: EventSource['onopen'];
+    onerror: EventSource['onerror'];
+    onmessage: EventSource['onmessage'];
+
+    close (): void;
+    addEventListener (inType: string, callback: (event: MessageEvent) => void): void;
+    removeEventListener (inType: string, callback: (event: MessageEvent) => void): void;
+}
```

Comment:
These typings should probably match the `EventSource` interface so that `ReconnectableEventSource` has complete compatibility.

```ts
interface EventSource extends EventTarget {
    onerror: ((this: EventSource, ev: Event) => any) | null;
    onmessage: ((this: EventSource, ev: MessageEvent) => any) | null;
    onopen: ((this: EventSource, ev: Event) => any) | null;
    /**
     * Returns the state of this EventSource object's connection. It can have the values described below.
     */
    readonly readyState: number;
    /**
     * Returns the URL providing the event stream.
     */
    readonly url: string;
    /**
     * Returns true if the credentials mode for connection requests to the URL providing the event stream is set to "include", and false otherwise.
     */
    readonly withCredentials: boolean;
    /**
     * Aborts any instances of the fetch algorithm started for this EventSource object, and sets the readyState attribute to CLOSED.
     */
    close(): void;
    readonly CLOSED: number;
    readonly CONNECTING: number;
    readonly OPEN: number;
    addEventListener<K extends keyof EventSourceEventMap>(type: K, listener: (this: EventSource, ev: EventSourceEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
    addEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions): void;
    removeEventListener<K extends keyof EventSourceEventMap>(type: K, listener: (this: EventSource, ev: EventSourceEventMap[K]) => any, options?: boolean | EventListenerOptions): void;
    removeEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | EventListenerOptions): void;
}
```

Reply:
Updated